src/Sarif.Driver/Sdk/AnalyzeCommandBase.cs

```diff
@@ -601,18 +603,38 @@ private SupportedPlatform GetCurrentRunningOS()
                     context.TargetUri.GetFileName()));
             }
 
-            if (options.ComputeFileHashes && _resultsCachingLogger.HashToResultsMap.TryGetValue(context.Hashes.Sha256, out List<Tuple<ReportingDescriptor, Result>> results))
+            if (options.ComputeFileHashes)
             {
-                context.Logger.AnalyzingTarget(context);
-                foreach (Tuple<ReportingDescriptor, Result> result in results)
+                _resultsCachingLogger.HashToResultsMap.TryGetValue(context.Hashes.Sha256, out List<Tuple<ReportingDescriptor, Result>> results);
+                _resultsCachingLogger.HashToNotificationsMap.TryGetValue(context.Hashes.Sha256, out List<Notification> notifications);
+
+                bool replayCachedData = (results != null || notifications != null);
+
+                if (replayCachedData)
                 {
-                    if (result.Item2.Locations?.Count > 0)
+                    context.Logger.AnalyzingTarget(context);
+
+                    if (results != null)
+                    {
+                        foreach (Tuple<ReportingDescriptor, Result> result in results)
```

Comment:
>result [](start = 68, length = 6)

Perhaps a different variable name, like `cachedResult` and `cachedResults` to distinguish these tupes and lists of tuples from `Result` objects.

Reply:
Nice, yes.

---
In reply to: [316890028](https://github.com/microsoft/sarif-sdk/pull/1658#discussion_r316890028) [](ancestors = 316890028)